spec/requests/circuit_breaker_spec.rb

```diff
@@ -0,0 +1,37 @@
+require "rails_helper"
+
+describe "Circuit breaker" do
+  let(:error) { GetIntoTeachingApiClient::CircuitBrokenError }
+
+  before do
+    api_constants = GetIntoTeachingApiClient.constants.select { |klass| klass.to_s.end_with?("Api") }
+    apis_to_mock = api_constants.map { |const| GetIntoTeachingApiClient.const_get(const) }
+    apis_to_mock.each do |api|
+      api.instance_methods(false).each do |method|
+        allow_any_instance_of(api).to receive(method).and_raise(error)
+      end
+    end
+  end
+
```

Comment:
I wanted to parameterise these, but wasn't sure how to pass in the route helpers

Reply:
You should be able to do it by setting up outside of the RSpec block and looping through the scenarios you want to test. There's an example in the https://github.com/DFE-Digital/get-into-teaching-app/blob/master/spec/requests/redirects_spec.rb